arch/risc-v/src/bl602/bl602_dma.h

```diff
@@ -0,0 +1,262 @@
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define BL602_DMA_NCHANNELS    4
+#define BL602_DMA_INT_EVT_TC  0 /* Interrupt callback status Transfer Complete */
+#define BL602_DMA_INT_EVT_ERR 1 /* Interrupt callback status Error */
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+#ifdef CONFIG_DEBUG_DMA_INFO
+struct bl602_dmaregs_s
+{
+  uint32_t intstatus;         /* Interrupt status. */
+  uint32_t inttcstatus;       /* Interrupt terminal count request status. */
+  uint32_t inttcclear;        /* Terminal count request clear. */
+  uint32_t interrorstatus;    /* Interrupt error status. */
+  uint32_t interrclr;         /* Interrupt error clear. */
+  uint32_t rawinttcstatus;    /* Status of the terminal count interrupt prior to masking. */
+  uint32_t rawinterrorstatus; /* Status of the error interrupt prior to masking. */
+  uint32_t enbldchns;         /* Channel enable status. */
+  uint32_t softbreq;          /* Software burst request. */
+  uint32_t softsreq;          /* Software single request. */
+  uint32_t softlbreq;         /* Software last burst request. */
+  uint32_t softlsreq;         /* Software last single request. */
+  uint32_t top_config;        /* DMA general configuration. */
+  uint32_t sync;              /* DMA request asynchronous setting. */
+  uint32_t c0srcaddr;         /* Channel DMA source address. */
+  uint32_t c0dstaddr;         /* Channel DMA Destination address. */
+  uint32_t c0lli;             /* Channel DMA link list. */
+  uint32_t c0control;         /* Channel DMA bus control. */
+  uint32_t c0config;          /* Channel DMA configuration. */
+  uint32_t c1srcaddr;         /* Channel DMA source address. */
+  uint32_t c1dstaddr;         /* Channel DMA Destination address. */
+  uint32_t c1lli;             /* Channel DMA link list. */
+  uint32_t c1control;         /* Channel DMA bus control. */
+  uint32_t c1config;          /* Channel DMA configuration. */
+  uint32_t c2srcaddr;         /* Channel DMA source address. */
+  uint32_t c2dstaddr;         /* Channel DMA Destination address. */
+  uint32_t c2lli;             /* Channel DMA link list. */
+  uint32_t c2control;         /* Channel DMA bus control. */
+  uint32_t c2config;          /* Channel DMA configuration. */
+  uint32_t c3srcaddr;         /* Channel DMA source address. */
+  uint32_t c3dstaddr;         /* Channel DMA Destination address. */
+  uint32_t c3lli;             /* Channel DMA link list. */
+  uint32_t c3control;         /* Channel DMA bus control. */
+  uint32_t c3config;          /* Channel DMA configuration. */
+};
+#endif
+
+/* Configuration for the LLI Control entry, defined as matching
+ * the DMA channel control register.
+ */
+
+begin_packed_struct struct bl602_dma_ctrl_s
```

Comment:
Do we really need packed for but fields?

Reply:
Re-read that part of the standard, and I think you are correct that it is safe to drop here.